src/segment-loader.js

```diff
@@ -503,14 +503,16 @@ export default class SegmentLoader extends videojs.EventTarget {
   }
 
   /**
-   * load a specific segment from a request into the buffer
+   * trim the back buffer so we only remove content
+   * on segment boundaries
    *
    * @private
+   *
+   * @param {Object} segmentInfo - the current segment
+   * @returns {Number} removeToTime - the end point in time, in seconds
+   * that the the buffer should be trimmed.
    */
-  loadSegment_(segmentInfo) {
-    let segment;
-    let keyXhr;
-    let segmentXhr;
+  trimBuffer_(segmentInfo) {
```

Comment:
I think a rebase pulled in something it should not have.. this code appears to be from the last commit in master.